com.unity.perception/Runtime/GroundTruth/PerceptionCamera.cs

```diff
@@ -50,6 +43,61 @@ public partial class PerceptionCamera : MonoBehaviour
         /// </summary>
         public Camera attachedCamera => m_AttachedCamera;
 
+        /// <summary>
+        /// Frame number at which this camera starts capturing.
+        /// </summary>
+        public int firstCaptureFrame = 0;
+
+        /// <summary>
+        /// Capture trigger modes for <see cref="PerceptionCamera"/>.
+        /// </summary>
+        public enum CaptureTriggerMode
```

Comment:
It is not good form to define enums inside classes if they are part of the public API. https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/nested-types

This looks like it should be moved out.

Reply:
Also, SimulationState should not depend on PerceptionCamera, since it should be able to be used by any sensor implementation.

Reply:
I moved the enum to DatasetCapture.cs, and out of any other class. Seems like a sensible place for it as sensor stuff is defined there.